Windows tests fail on file cleanup

### Component(s)

pkg/stanza

### What happened?

Tests fail with on Windows:
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5756861995/job/15610273677?pr=24880

```
--- FAIL: TestFinder (21.22s)
    --- FAIL: TestFinder/Timestamp_Sorting (1.46s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderTimestamp_Sorting2067807901\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Timestamp_Sorting_Ascending (1.48s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderTimestamp_Sorting_Ascending980688751\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Numeric_Sorting (1.69s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderNumeric_Sorting2945291070\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Numeric_Sorting_Ascending (1.93s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderNumeric_Sorting_Ascending1997561379\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Alphabetical_Sorting (1.74s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderAlphabetical_Sorting2331360315\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Alphabetical_Sorting_Ascending (1.96s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderAlphabetical_Sorting_Ascending3571327606\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Multiple_Sorting_-_timestamp_priority_sort (2.01s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderMultiple_Sorting_-_timestamp_priority_sort1452414392\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Multiple_Sorting_-_timestamp_priority_sort_-_numeric_ascending (1.72s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderMultiple_Sorting_-_timestamp_priority_sort_-_numeric_ascending2900057276\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Multiple_Sorting_-_timestamp_priority_sort#01 (1.51s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderMultiple_Sorting_-_timestamp_priority_sort#013300103737\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Multiple_Sorting_-_alpha_priority_sort_-_alpha_ascending (1.77s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderMultiple_Sorting_-_alpha_priority_sort_-_alpha_ascending1904492045\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Multiple_Sorting_-_alpha_priority_sort_-_timestamp_ascending (2.04s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderMultiple_Sorting_-_alpha_priority_sort_-_timestamp_ascending192357911\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Multiple_Sorting_-_alpha_priority_sort_-_timestamp_ascending#01 (1.91s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderMultiple_Sorting_-_alpha_priority_sort_-_timestamp_ascending#01488380667\001: The process cannot access the file because it is being used by another process.
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer	28.004s
?   	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/emit	[no test files]
--- FAIL: TestFinder (25.52s)
    --- FAIL: TestFinder/IncludeOne (2.01s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderIncludeOne698495821\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/IncludeNone (1.50s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderIncludeNone1606326002\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/IncludeAll (1.89s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderIncludeAll3066724630\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/IncludeLogs (1.84s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderIncludeLogs915139469\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/IncludeA (1.44s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderIncludeA1308524544\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Include2s (1.91s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderInclude2s948130268\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/Exclude (1.90s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderExclude543302135\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/ExcludeMany (2.00s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderExcludeMany662117668\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/ExcludeDuplicates (1.46s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderExcludeDuplicates745031694\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/IncludeMultipleDirectories (1.69s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderIncludeMultipleDirectories1384715364\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/IncludeMultipleDirectoriesVaryingDepth (1.63s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderIncludeMultipleDirectoriesVaryingDepth3105878215\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/DoubleStarSameDepth (1.78s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderDoubleStarSameDepth31501[740](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5756861995/job/15610273677?pr=24880#step:6:741)69\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/DoubleStarVaryingDepth (1.69s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderDoubleStarVaryingDepth511166870\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/SingleLevelFilesOnly (1.47s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderSingleLevelFilesOnly543269683\001: The process cannot access the file because it is being used by another process.
    --- FAIL: TestFinder/MultiLevelFilesOnly (1.32s)
        testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestFinderMultiLevelFilesOnly3333458789\001: The process cannot access the file because it is being used by another process.
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/finder	25.555s
```

### Collector version

latest

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go
+++ b/pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go
@@ -154,13 +154,13 @@ func TestFindFiles(t *testing.T) {
 		},
 		{
 			name:     "SingleLevelFilesOnly",
-			files:    []string{"a1.log", "a2.txt", "b/b1.log", "b/b2.txt"},
+			files:    []string{"a1.log", "a2.txt", filepath.Join("b", "b1.log"), filepath.Join("b", "b2.log")},
 			include:  []string{"*"},
 			expected: []string{"a1.log", "a2.txt"},
 		},
 		{
 			name:     "MultiLevelFilesOnly",
-			files:    []string{"a1.log", "a2.txt", "b/b1.log", "b/b2.txt", "b/c/c1.csv"},
+			files:    []string{"a1.log", "a2.txt", filepath.Join("b", "b1.log"), filepath.Join("b", "b2.txt"), filepath.Join("b", "c", "c1.csv")},
 			include:  []string{filepath.Join("**", "*")},
 			expected: []string{"a1.log", "a2.txt", filepath.Join("b", "b1.log"), filepath.Join("b", "b2.txt"), filepath.Join("b", "c", "c1.csv")},
 		},
@@ -168,10 +168,21 @@ func TestFindFiles(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
+			cwd, err := os.Getwd()
+			require.NoError(t, err)
 			require.NoError(t, os.Chdir(t.TempDir()))
+			defer func() {
+				require.NoError(t, os.Chdir(cwd))
+			}()
 			for _, f := range tc.files {
 				require.NoError(t, os.MkdirAll(filepath.Dir(f), 0700))
-				require.NoError(t, os.WriteFile(f, []byte(filepath.Base(f)), 0000))
+
+				file, err := os.OpenFile(f, os.O_CREATE|os.O_RDWR, 0600)
+				require.NoError(t, err)
+
+				_, err = file.WriteString(filepath.Base(f))
+				require.NoError(t, err)
+				require.NoError(t, file.Close())
 			}
 			assert.Equal(t, tc.expected, FindFiles(tc.include, tc.exclude))
 		})

--- a/pkg/stanza/fileconsumer/matcher/matcher.go
+++ b/pkg/stanza/fileconsumer/matcher/matcher.go
@@ -72,23 +72,23 @@ func New(c Criteria) (*Matcher, error) {
 	for _, sc := range c.OrderingCriteria.SortBy {
 		switch sc.SortType {
 		case sortTypeNumeric:
-			s, err := filter.SortNumeric(sc.RegexKey, sc.Ascending)
+			f, err := filter.SortNumeric(sc.RegexKey, sc.Ascending)
 			if err != nil {
 				return nil, fmt.Errorf("numeric sort: %w", err)
 			}
-			filterOpts = append(filterOpts, s)
+			filterOpts = append(filterOpts, f)
 		case sortTypeAlphabetical:
-			s, err := filter.SortAlphabetical(sc.RegexKey, sc.Ascending)
+			f, err := filter.SortAlphabetical(sc.RegexKey, sc.Ascending)
 			if err != nil {
 				return nil, fmt.Errorf("alphabetical sort: %w", err)
 			}
-			filterOpts = append(filterOpts, s)
+			filterOpts = append(filterOpts, f)
 		case sortTypeTimestamp:
-			s, err := filter.SortTemporal(sc.RegexKey, sc.Ascending, sc.Layout, sc.Location)
+			f, err := filter.SortTemporal(sc.RegexKey, sc.Ascending, sc.Layout, sc.Location)
 			if err != nil {
 				return nil, fmt.Errorf("timestamp sort: %w", err)
 			}
-			filterOpts = append(filterOpts, s)
+			filterOpts = append(filterOpts, f)
 		default:
 			return nil, fmt.Errorf("'sort_type' must be specified")
 		}

--- a/pkg/stanza/fileconsumer/matcher/matcher_test.go
+++ b/pkg/stanza/fileconsumer/matcher/matcher_test.go
@@ -539,10 +539,20 @@ func TestMatcher(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
+			cwd, err := os.Getwd()
+			require.NoError(t, err)
 			require.NoError(t, os.Chdir(t.TempDir()))
+			defer func() {
+				require.NoError(t, os.Chdir(cwd))
+			}()
 			for _, f := range tc.files {
 				require.NoError(t, os.MkdirAll(filepath.Dir(f), 0700))
-				require.NoError(t, os.WriteFile(f, []byte(filepath.Base(f)), 0000))
+				file, fErr := os.OpenFile(f, os.O_CREATE|os.O_RDWR, 0600)
+				require.NoError(t, fErr)
+
+				_, fErr = file.WriteString(filepath.Base(f))
+				require.NoError(t, fErr)
+				require.NoError(t, file.Close())
 			}
 			matcher, err := New(Criteria{
 				Include:          tc.include,

